skimage/feature/corner.py

```diff
@@ -67,41 +67,59 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0):
     cval : float, optional
         Used in conjunction with mode 'constant', the value outside
         the image boundaries.
+    order : {'rc', 'xy'}, optional
+        This parameter allows for the use of reverse or forward order of
+        the image axes in gradient computation. 'rc' indicates the use of
+        the first axis initially (Arr, Arc, Acc), whilst 'xy' indicates the
+        usage of the last axis initially (Axx, Axy, Ayy).
 
     Returns
     -------
-    Axx : ndarray
+    Arr : ndarray
         Element of the structure tensor for each pixel in the input image.
-    Axy : ndarray
+    Arc : ndarray
         Element of the structure tensor for each pixel in the input image.
-    Ayy : ndarray
+    Acc : ndarray
         Element of the structure tensor for each pixel in the input image.
 
     Examples
     --------
     >>> from skimage.feature import structure_tensor
     >>> square = np.zeros((5, 5))
     >>> square[2, 2] = 1
-    >>> Axx, Axy, Ayy = structure_tensor(square, sigma=0.1)
-    >>> Axx
+    >>> Arr, Arc, Acc = structure_tensor(square, sigma=0.1, order='rc')
+    >>> Acc
     array([[0., 0., 0., 0., 0.],
            [0., 1., 0., 1., 0.],
            [0., 4., 0., 4., 0.],
            [0., 1., 0., 1., 0.],
            [0., 0., 0., 0., 0.]])
 
     """
+    if order is None:
+        if image.ndim == 2:
+            # The legacy 2D code followed (x, y) convention, so we swap the
+            # axis order to maintain compatibility with old code
+            warn('deprecation warning: the default order of the structure '
```

Comment:
```suggestion
            FutureWarning('The default order of the structure '
```

Reply:
I think this should be a FutureWarning? In the last sentence you may want to add (Specify order="xy" to maintain the old behavior).

Reply:
(it is fine if you want to edit in a separate commit instead of merging the suggestion here)

Reply:
I added the note at the end, but I didn't understand the Suggested change. The category of the warning is already set to `FutureWarning`. Did you want me to raise the warning instead of using `warn`?

Reply:
Ah, nevermind. I had missed the category argument here. I will just close this as resolved